src/CarbonAware.LocationSources/CarbonAware.LocationSources.Azure/test/AzureLocationSourceTest.cs

```diff
@@ -36,10 +36,10 @@ public async Task TestToGeopositionLocation_ValidLocation()
     }
 
     // <summary>
-    // If an Azure Location with invalid RegionName is passed, should fail.
+    // If an Azure Location with no LocationType is passed, should fail.
     // </summary>
     [Test]
-    public void TestToGeopositionLocation_InvalidLocation()
+    public void TestToGeopositionLocation_LocationTypeNotProvided_ThrowsException()
```

Comment:
I would make RegionName a "valid" region name in this test, or at least something that looks valid, because if I hadn't read the title, just by looking at the code I expect that this test is testing whether an `invalid location` throws an error, not the lack of `LocationType`. And it makes me wonder about "order of operations" where-in do we fail first because of lack of the `LocationType` or because of the invalid name